torch/csrc/jit/pybind_utils.h

```diff
@@ -170,6 +234,18 @@ inline InferredType tryToInferContainerType(py::handle input) {
         return type_match.reason();
       }
     }
+    py::object input_type =py::cast<py::object>(input.get_type());
+    if (isNamedTupleClass(input_type)) {
+      // If the tuple is a NamedTuple type
+      auto qualifiedName = c10::QualifiedName(py::cast<std::string>(
+          py::module::import("torch.jit").attr("_qualified_name")(input.get_type())));
+      c10::TypePtr named_type = getOrCreateNamedTupleType(
+        input_type,
+        qualifiedName,
+        element_types,
+        tracer::getPythonInterpreterSourceRange());
```

Comment:
I am not sure `tracer::getPythonInterpreterSourceRange()` should be the way to get the source range here, but I think it's acceptable since we always have python interpreter here.